deployment/profiled-injector.yml

```diff
@@ -1,6 +1,30 @@
 ---
-- hosts: injector
-  become: yes
-  roles:
-    - profiled-traffic-generator
-    - injector-configurator
+- hosts:
```

Comment:
cannot be reused? same as injector.yaml

Reply:
`profiled-traffic-generator` is [reused](https://github.com/whitingjr/perftest-toolkit/blob/a5a0909e8b7e54bff2c0b236c1618978b2e73daa/deployment/profiled-injector.yml#L6).
I don't see keeping the role name `injector-configurator` can still be relevant when the prior role is customized for building and configuring JMeter. Instead Hyperfoil is installed [here](https://github.com/whitingjr/perftest-toolkit/blob/a5a0909e8b7e54bff2c0b236c1618978b2e73daa/deployment/profiled-injector.yml#L30) and configured [here](https://github.com/whitingjr/perftest-toolkit/blob/a5a0909e8b7e54bff2c0b236c1618978b2e73daa/deployment/profiled-injector.yml#L22). Renaming the role to `injector-setup` better communicates the purpose of the role.

Reply:
Both profiled-injector.yml and injector.yml are 30 lines files. The only differ in one line:

```
$ diff profiled-injector.yml injector.yml 
6c6
<       name: profiled-traffic-generator
---
>       name: user-traffic-reader
```
I am pretty sure that ansible provides ways to avoid DRY

Reply:
DRY

Reply:
That certainly can be re-factored to avoid repeating commands.

Reply:
What do you think of switching between the 2 roles using the value of a variable ?
```
---
# variables file all.yml

# Traffic source has two values, user or profiled
traffic_source: profiled
```
```
- hosts:
    - hyperfoil_controller
  tasks:
  - include_role:
      name: profiled-traffic-generator
    when: traffic_source == 'profiled'
- hosts:
    - hyperfoil_controller
  tasks:
  - include_role:
      name: user-traffic-reader
    when: traffic_source == 'user'

```

Reply:
I think there should exists two playbooks. The roles are reusable pieces. The playbook defines some sequence of roles to be executed. Do not specify tasks in playbooks and it should be fine.

Reply:
Ok, good to know.